contracts/badger-bridge/BadgerBridgeAdapter.sol

```diff
@@ -96,57 +98,87 @@ contract BadgerBridgeAdapter is OwnableUpgradeable, ReentrancyGuardUpgradeable {
         // user args
         address _token, // either renBTC or wBTC
         uint256 _slippage,
-        address payable _destination,
+        address _user,
+        address _vault,
         // darknode args
         uint256 _amount,
         bytes32 _nHash,
         bytes calldata _sig
     ) external nonReentrant {
-        require(_token == address(renBTC) || _token == address(wBTC), "invalid token address");
+        require(_token == address(renBTC) || _token == address(wBTC) || approvedVaults[_vault], "invalid token address");
```

Comment:
This will revert every single call if `vault` is not set (which it won't be in the renbtc/wbtc case).